### PR TITLE
Add support for dbt-duckdb to the dbt_date package

### PR DESCRIPTION
**Pull request by jwills**
Hello hello-- had some users testing out the new dbt semantic layer example with dbt-duckdb and we noticed some deltas between what DuckDB supports and the default macros that seemed worth fixing up; if you would prefer I do this via a shim package, please let me know! 🙇

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -65,6 +65,12 @@ jobs:
             . venv/bin/activate
             dbt build -t snowflake --project-dir $DBT_PROJECT_DIR
 
+      - run:
+          name: "Run Tests - DuckDB"
+          command: |
+            . venv/bin/activate
+            dbt build -t duckdb --project_dir $DBT_PROJECT_DIR
```
Comment (clausherther) [inline .circleci/config.yml]: Hmmm, somehow CI failed on this line https://app.circleci.com/pipelines/github/calogica/dbt-date/83/workflows/f4415d81-0620-43a2-93e1-f0303c64f2f9/jobs/58?invite=true#step-108-27

**Review by clausherther — CHANGES_REQUESTED** [review]
Hi @jwills! Normally, I would have said let's do this in a shim package, but then you went ahead and added CI support making this very persuasive 😄 
If we can the CI bit figured out, I think this should be good, but I'll give it a closer look then. Thanks for this!

File: .circleci/config.yml (hunk 1 above)
Comment (jwills) [reply .circleci/config.yml]: A silly typo-- fixed!

**Comment by jwills** [comment]
okay, equally silly error b/c the integration tests are running an old version of dbt-duckdb (1.4.1) instead of the latest that I ran locally-- will fix shortly

**Comment by clausherther** [comment]
> okay, equally silly error b/c the integration tests are running an old version of dbt-duckdb (1.4.1) instead of the latest that I ran locally-- will fix shortly

Looks like that did it!

**Comment by clausherther** [comment]
Would you mind also updating the README to note the support for duckdb?

I will have probably have to release this is a 0.8.0 which also means a bump to the dbt-expectations version so might take a bit.

**Review by clausherther — APPROVED** [review]
👍

**Pull request merged**